src/community/label_propagation.jl

```diff
@@ -0,0 +1,106 @@
+"""
+Community detection using the label propagation algorithm (see [Raghavan et al.](http://arxiv.org/abs/0709.2938)).
+`g`: imput Graph
+`maxiter`: maximum number of iterations
+return : vertex assignments and the convergence history
+"""
+function label_propagation(g::SimpleGraph; maxiter=1000)
+    n = nv(g)
+    label = collect(1:n)
+    runing_nodes = Set(vertices(g))
+    c = NeighComm(collect(1:n), fill(-1,n), 1)
+    convergence_hist = Vector{Int}()
+    random_order = Array(Int, n)
+    i = 0
+    while !isempty(runing_nodes) && i < maxiter
+        num_active = length(runing_nodes)
+        push!(convergence_hist, num_active)
+        i += 1
+        
+        # processing nodes in random order
+        for (j,node) in enumerate(runing_nodes)
+            random_order[j] = node
+        end
+        range_shuffle!(1:num_active, random_order)
+        @inbounds for j=1:num_active
+            u = random_order[j]
+            old_comm = label[u]
+            label[u] = vote!(g, label, c, u)
+            if old_comm != label[u]
+                for v in out_neighbors(g, u)
+                    push!(runing_nodes, v)
```

Comment:
You might be able to do this more efficiently with `append!(running_nodes, out_neighbors(g,u))`